CGMBLEKit Example/ViewController.swift

```diff
@@ -62,6 +62,36 @@ class ViewController: UIViewController, TransmitterDelegate, UITextFieldDelegate
         UserDefaults.standard.passiveModeEnabled = sender.isOn
     }
 
+    @IBAction func calibrate(_ sender: UIButton) {
+        let dialog = UIAlertController(title: "Enter BG", message: "Calibrate sensor.", preferredStyle: .alert)
+
+        let unit = UserDefaults.standard.glucoseUnit
+
+        dialog.addTextField { (textField : UITextField!) in
+            textField.placeholder = unit.unitString
+            textField.keyboardType = .numberPad
+        }
+
+        dialog.addAction(UIAlertAction(title: "Calibrate", style: .default, handler: { (action: UIAlertAction!) in
+            let textField = dialog.textFields![0] as UITextField
+            let minGlucose = HKQuantity(unit: HKUnit.milligramsPerDeciliter(), doubleValue: 40)
+            let maxGlucose = HKQuantity(unit: HKUnit.milligramsPerDeciliter(), doubleValue: 400)
+
+            if let text = textField.text, let entry = Double(text) {
+                guard entry >= minGlucose.doubleValue(for: unit) && entry <= maxGlucose.doubleValue(for: unit) else {
+                    // TODO: notify the user if the glucose is not in range
+                    return
+                }
+                let glucose = HKQuantity(unit: unit, doubleValue: Double(entry))
+                AppDelegate.sharedDelegate.commandQueue.enqueue(.calibrateSensor(to: glucose, at: Date()))
```

Comment:
It's not safe to mutate `commandQueue` here on the main thread, and also mutate it on a background thread during `TransmitterDelegate.dequeuePendingCommand(for:)`